pandas/core/indexes/base.py

```diff
@@ -1672,7 +1672,7 @@ def _drop_level_numbers(self, levnums: List[int]):
         Drop MultiIndex levels by level _number_, not name.
         """
 
-        if not levnums:
+        if not levnums and (len(self) > 1 or not isinstance(self, ABCMultiIndex)):
```

Comment:
this len(self) > 1 is a very odd check, what is the intent?

Reply:
Seems to be no longer necessary. Code changed quite a bit since I implemented this. Removed it.